File: 0344-reverse-string/0344-reverse-string.dart
//inPlace means doing operation on the same array without copping it
class Solution {
  void reverseString(List<String> s) {
    int l = 0;
    int r = s.length  - 1;
    if(s.length == 1) return;
    while (l < r) {
      String temp = s[l];
      s[l] = s[r];
      s[r] = temp;
      l = l + 1;
      r = r - 1;
    }
    //print(s);
  }
}

File: 0344-reverse-string/0344-reverse-string.py
class Solution:
    def reverseString(self, s: list[str]) -> None:
        
        left = 0
        right = len(s) - 1
        while(left <= right):
            # Tuple Unpacking
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1